# Formulle d'un crédit d'immobilier

#### Exemple
Soit par exemple un crédit immobilier d'un montant de $200 000$ €, emprunté sur $20$ ans à un taux d’intérêt de $1 \%$, soit $240$ mensualités de $919,79 €$.	

Lors du paiement de la première mensualité, l'emprunteur doit encore 200 000 € à la banque. Les intérêts mensuels s’élèvent à un douzième des intérêts annuels sur cette somme, soit $(200 000 x 1 \%) / 12 = 166,67$ €.

La première mensualité, d’un montant total de $919,79$ €, se compose donc de $166,67$ € d’intérêts et de $753,12 €$ de remboursement de capital.

Le mois suivant, le capital restant dû diminue de $753,12$ €. Il ne s’élève donc plus qu'à $199 246,88$ €.

La mensualité suivante se composera donc que de $(199 246,88 x 1 \%) / 12 = 166,04 €$ d'intérêts, soit légèrement moins que le mois précédent. Le capital remboursé passe à $753,75= (919,79 – 166,04)$.
#### Problème
Soit un crédit $C$ pendant $N$ mois pour le taux de $T\%$ par mois. Calculer la mensualité de ce crédit. 

On note $M$ la mensualité, et $C_n$ le capital restant au $k$-ième mois. On obtient alors l'équation récurrente
	$$C_{k+1}=C_n-(M-tC_k),$$
équivaut 
	  $$C_{k+1}-\frac{M}{t}=(1+M)\left(C_k-\frac{M}{t}\right),$$
ça nous dit que la suite $\left(C_k-\frac{M}{t}\right) $ est une suite géométrique, donc 
	$$C_k-\frac{M}{T}=(1+T)^k \left(C_0-\frac{M}{T}\right).$$

Par ailleurs, $C_N=0$, on obtient alors l'égalité suivante
		$$\frac{M}{T}+(1+T)^k \left(C_0-\frac{M}{T} \right)=0,$$
ça nous donne que 
		$$M=\frac{TC(1+T)^{N}}{(1+T)^{N}-1} .$$

In [1]:
import numpy as np

In [6]:
def interet(C,t,n):
    #n: Durée de crédit (ans).
    #t: taux par un.
    N=n*12
    T=t/12
    const=(1+T)**N
    M=C*T*const/(const-1)
    Int=M*N-C
    #Int: Total des intérêts
    return (M, Int)

In [11]:
interet(250000,0.038,25)

(1292.1414042268157, 137642.42126804468)

Pour un crédit 250000 euros, le taux de 3,8% par un pendant 25 ans, alors la mensualité est 1292,14 euros et le total des intérêt est 137642.42 euros.